core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -208,46 +214,39 @@ class BillingProjectOrchestrator(ctx: RawlsRequestContext,
         case Some(_) => bpmBillingProjectLifecycle
       }
       _ <- billingRepository.failUnlessHasNoWorkspaces(projectName)
-      _ <- projectLifecycle.preDeletionSteps(projectName, ctx)
-      _ <- unregisterBillingProjectWithUserInfo(projectName, ctx.userInfo)
-    } yield {}
+      (jobControlId, monitorJobType) <- projectLifecycle.initiateDelete(projectName, ctx)
+      _ <- resourceMonitorRecordDao.create(
```

Comment:
If we wanted to keep the synchronous behavior for GCP projects for now, we would either match on the monitorJobType, or return an additional value when we choose the lifecycle, and then for GCP projects, call the finalizeDeletion step, instead of scheduling a job.